src/cmd/usb.js

```diff
@@ -186,52 +187,51 @@ module.exports = class UsbCommand extends CLICommandBase {
 			.then(() => console.log('Done.'));
 	}
 
-	cloudStatus(args, started = Date.now()){
+	async cloudStatus(args) {
 		const { until, timeout, params: { device } } = args;
+		this.newSpin('Querying device...').start();
+		let status = null;
 
-		if (Date.now() - (started + timeout) > 0){
-			throw new Error('timed-out waiting for status...');
+		try {
+			status = await executeWithUsbDevice({
+				args: { idOrName: device }, // device here is the id
+				func: (dev) => this._cloudStatus(dev, until, timeout),
+			});
+		} catch (error) {
+			this.stopSpin();
+			throw new Error(error.message);
+		} finally {
+			this.stopSpin();
```

Comment:
`finally` will always be executed, so you can remove `this.stopSpin()` from catch since you are doing it in the `finally` block

Check here 👉  [try-catch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)

Reply:
I wanted to ensure the error thrown does not interfere with the spinner

Reply:
It shouldn't since `finally` will always be executed. 
![image](https://github.com/user-attachments/assets/a9958fd9-76b0-4b06-8e63-519bd3ab16ab)